app/helpers/elmo_form_builder.rb

```diff
@@ -132,7 +132,13 @@ def elmo_field(field_name, options)
       if options[:read_only]
 
         # get field value
-        val = @object.send(field_name)
+        # audio_prompt is the audio file being set by paperclip
+        # to access the file name, we have to target the original_filename method
+        val = if field_name == :audio_prompt && @object.audio_prompt
+                @object.audio_prompt.original_filename
+              else
+                @object.send(field_name)
+              end
```

Comment:
BTW Jay there is a big problem with this approach. Can you think about it a bit before we meet and see if you can identify the issue?

Reply:
One thing that comes to mind is that this helper is being used in multiple places
Another thing I can think about is that the code is not flexible

Reply:
Both are correct. It's important to think about it in terms of "knowledge" of the class. Metz talks about this a lot. This class shouldn't have to "know" anything about a field called "audio_prompt". That's a "bad coupling". The class is a generic form builder that should work with any form, not just the specific one we're dealing with.

So as a rule, to have something as specific as this in a general utility class is a big red flag.

Can you rewrite it so that it works for _any_ file input?

You know it's safe to do that (won't break any existing functionality) because _you_ added the file field type in this issue. So that is nice!